server/app/models/ApplicationModel.java

```diff
@@ -177,6 +187,15 @@ void setLatestStatusForTest(String latestStatus) {
     this.latestStatus = latestStatus;
   }
 
+  /**
+   * This is visible only for tests to manipulate the latest note directly in order to ensure that
+   * updates to it are overridden by the configured database trigger.
+   */
+  // @VisibleForTesting
+  public void setLatestNote(String applicationNote) {
+    this.latestNote = applicationNote;
+  }
```

Comment:
I'd remove this, or make it only `@VisibleForTesting`, and enforce that the only way to set the latest note is via a method on the `ApplicationEventsRepository` within the same transaction as the `ApplicantEvent` is created.

Reply:
I have removed this method.